config/settings/production.yml

```diff
@@ -0,0 +1 @@
+basic_auth_required: true
```

Comment:
Is it not better to set this in `config/settings.yml`? In development, we change setting to false. That way we can ensure tests and prod use basic auth.

Reply:
By "tests" do you mean QA environments and not literally the Rails test env? We probably don't want basic auth for that when running tests.

Reply:
Also, this feels like it should be a "feature". In publish we have a [section in the settings for features](https://github.com/DFE-Digital/publish-teacher-training/blob/master/config/settings.yml#L63-L76) and a [feature service](https://github.com/DFE-Digital/publish-teacher-training/blob/master/app/services/feature_service.rb) to use them.

Is this the right time to look at adding something like this here?

Reply:
@misaka Personally I wouldn't really peg this as a feature. Seems more like an operational requirement similar to setting up CI and other boilerplate stuff. But happy to add the feature service in a separate follow up PR if you want?

Reply:
@aliuk2012 done in https://github.com/DFE-Digital/register-trainee-teacher-data/pull/32/commits/ca20ad5bba85c5d153f53e71e9ab527259a760da